File: apps/api/momento_api/models/share.py
from typing import Optional

from pydantic import BaseModel


class ShareLinkResponse(BaseModel):
    id: int
    token: str
    media_id: Optional[int]
    album_id: Optional[int]
    has_password: bool
    expires_at: Optional[str]
    view_count: int
    created_at: str


class ShareCreateRequest(BaseModel):
    media_id: Optional[int] = None
    album_id: Optional[int] = None
    password: Optional[str] = None
    expires_in_days: Optional[int] = None


class ShareDeleteRequest(BaseModel):
    share_id: int


class ShareListResponse(BaseModel):
    shares: list[ShareLinkResponse]


class ShareVerifyRequest(BaseModel):
    password: str

File: apps/api/src/models/share.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ShareLinkResponse {
    pub id: i64,
    pub token: String,
    pub media_id: Option<i64>,
    pub album_id: Option<i64>,
    pub has_password: bool,
    pub expires_at: Option<String>,
    pub view_count: i64,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ShareCreateRequest {
    pub media_id: Option<i64>,
    pub album_id: Option<i64>,
    pub password: Option<String>,
    pub expires_in_days: Option<i32>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ShareDeleteRequest {
    pub share_id: i64,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ShareListResponse {
    pub shares: Vec<ShareLinkResponse>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ShareVerifyRequest {
    pub password: String,
}
